src/transactions/TransactionFrame.cpp

```diff
@@ -1943,29 +1945,30 @@ TransactionFrame::processPostApply(Application& app,
                                    AbstractLedgerTxn& ltxOuter,
                                    TransactionMetaFrame& meta)
 {
-    processPostApply(app, ltxOuter, meta, getSourceID());
+    processRefund(app, ltxOuter, meta, getSourceID());
 }
 
 // This is a TransactionFrame specific function that should only be used by
 // FeeBumpTransactionFrame to forward a different account for the refund.
-void
-TransactionFrame::processPostApply(Application& app,
-                                   AbstractLedgerTxn& ltxOuter,
-                                   TransactionMetaFrame& meta,
-                                   AccountID const& feeSource)
+int64_t
+TransactionFrame::processRefund(Application& app, AbstractLedgerTxn& ltxOuter,
```

Comment:
Can this be `const` function?

Reply:
No it updates state (feeCharged).

Reply:
Hmm, maybe we could make these const and do `feeCharged -= refundSorobanFee()`. This way only `processPostApply` would be non-const. Admittedly, that doesn't help much with the footguns, so up to you.

Reply:
`loadAccount`, which is also called in `refundSorobanFee`, can't be marked `const`, so i don't think reworking any of this is worth the effort.